packages/util/src/remove-slashes.ts

```diff
@@ -11,12 +11,18 @@ export function removeSlashes(path: string): string {
  * @internal
  */
 export function removeTrailingSlashes(path: string): string {
-  return path.endsWith('/') ? path.slice(0, -1) : path;
+  while (path.endsWith('/')) {
+    path = path.endsWith('/') ? path.slice(0, -1) : path;
+  }
+  return path;
 }
 
 /**
  * @internal
  */
 export function removeLeadingSlashes(path: string): string {
-  return path.startsWith('/') ? path.slice(1) : path;
+  while (path.startsWith('/')) {
+    path = path.startsWith('/') ? path.slice(1) : path;
+  }
+  return path;
```

Comment:
Drive by change, to also account for multiple leading or trailing slashes.

Reply:
We decided to do either a name change or implementation change for v4, as this is a breaking change.